lib/web_ui/dev/test_platform.dart

```diff
@@ -0,0 +1,888 @@
+              // Precompiled directories often contain symlinks
+              serveFilesOutsidePath:
+                  config.suiteDefaults.precompiledPath != null))
+          .add(_wrapperHandler);
+    }
+
+    var pipeline = shelf.Pipeline()
+        .addMiddleware(PathHandler.nestedIn(_secret))
+        .addHandler(cascade.handler);
+
+    _server.mount(shelf.Cascade()
+        .add(createFileHandler(faviconPath))
+        .add(pipeline)
+        .handler);
+  }
+
+  Future<shelf.Response> _screeshotHandler(shelf.Request request) async {
+    if (!request.requestedUri.path.endsWith('/screenshot')) {
+      return shelf.Response.notFound(
+          'This request is not handled by the screenshot handler');
+    }
+
+    final String payload = await request.readAsString();
+    final Map<String, dynamic> requestData = json.decode(payload);
+    final String filename = requestData['filename'];
+    final bool write = requestData['write'];
+    final String result = await _diffScreenshot(filename, write);
+    return shelf.Response.ok(json.encode(result));
+  }
+
+  Future<String> _diffScreenshot(String filename, bool write) async {
+    const String _kGoldensDirectory = 'test/golden_files';
+    final wip.ChromeConnection chromeConnection =
+        wip.ChromeConnection('localhost', _kChromeDevtoolsPort);
+    final wip.ChromeTab chromeTab = await chromeConnection.getTab(
+        (wip.ChromeTab chromeTab) => chromeTab.url.contains('localhost'));
+    final wip.WipConnection wipConnection = await chromeTab.connect();
+    final wip.WipResponse response =
+        await wipConnection.sendCommand('Page.captureScreenshot');
+    final Uint8List bytes = base64.decode(response.result['data']);
+    final File file = File(p.join(_kGoldensDirectory, filename));
+    if (write) {
+      file.writeAsBytesSync(bytes, flush: true);
+      return 'Golden file $filename was updated. You can remove "write: true" in the call to matchGoldenFile.';
+    }
+    if (!file.existsSync()) {
+      return '''
+  Golden file $filename does not exist.
+
+  To automatically create this file call matchGoldenFile('$filename', write: true).
+  ''';
+    }
+    final List<int> goldenBytes = file.readAsBytesSync();
+    final int lengths = bytes.length;
+    for (int i = 0; i < lengths; i++) {
+      if (goldenBytes[i] != bytes[i]) {
+        if (write) {
+          file.writeAsBytesSync(bytes, flush: true);
+          return 'Golden file $filename was updated. You can remove "write: true" in the call to matchGoldenFile.';
+        } else {
+          final File failedFile = File(p.join(file.parent.path, '.${p.basename(file.path)}'));
+          failedFile.writeAsBytesSync(bytes, flush: true);
+
+          // TODO(yjbanov): do not fail Cirrus builds. They currently fail because Chrome produces
+          //                different pictures. We need to pin Chrome versions and use a fuzzy image
+          //                comparator.
```

Comment:
I would look at what flutter_goldens is doing with package:image

Reply:
Good idea. Looking at it right now.